src/main/java/com/checkmarx/intellij/tool/window/actions/StartScanAction.java

```diff
@@ -315,4 +322,8 @@ private static String msg(Resource resource, Object... params) {
     public @NotNull ActionUpdateThread getActionUpdateThread() {
         return ActionUpdateThread.BGT;
     }
+
+    private String getActiveBranch(Project project) {
+        return Utils.getRootRepository(project) == null ? null : Objects.requireNonNull(Utils.getRootRepository(project)).getCurrentBranchName();
```

Comment:
If this will be null: Utils.getRootRepository(project)).getCurrentBranchName() Objects.requireNonNull will throw exception that we will not catch.
maybe using Objects.requireNonNullElse is better here

Reply:
but what is the meaning of a SCM project that not connect to any branch?
how can we have repository without any branch connected to?

Reply:
does this Utils.getRootRepository(project)).getCurrentBranchName() can throw exception? If so we must catch it, else the plugin will break

Reply:
pay attention that if Utils.getRootRepository(project)) is null we will return null and not continue
and about the other part - method getCurrentBranchName isn't throwing any exception